soda_machine.rb

```diff
@@ -7,12 +9,54 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+      @sodas.length
   end
 
+  # This is currently not working properly.
+  # It works for the #find_soda tests,
+  # But does not return properly in the #sell method.
   def find_soda(soda_brand)
+    @sodas.each do |soda|
+      if soda.brand == soda_brand
+        return soda
+      else
+        return nil
```

Comment:
The reason this isn't working out is a control flow logic error.

if the soda matches return the soda else return nil. So it only checks the first soda in the array and then returns something. The return nil should be placed outside the each loop and that will resolve your issue.